posts/2019-05-23-Rust-1.35.0.md

```diff
@@ -0,0 +1,290 @@
+- [`RangeTo::contains`]
+- [`RangeInclusive::contains`]
+- [`RangeToInclusive::contains`]
+
+With these, you can easily check whether a given value exists in a range. For example, you may write:
+
+```rust
+fn main() {
+    if (0..=10).contains(&5) {
+        println!("Five is included in zero to ten.");
+    }
+}
+```
+
+#### Map and split a borrowed `RefCell` value in two
+
+With Rust 1.35.0, you can now map and split the borrowed value of a `RefCell` into multiple borrows for different components of the borrowed data:
+
+- [`Ref::map_split`]
+- [`RefMut::map_split`]
+
+#### Replace the value of a `RefCell` through a closure
+
+This release introduces a convenience method `replace_with` on `RefCell`:
+
+- [`RefCell::replace_with`]
+
+With it, you can more ergonomically map and replace the current value of the cell and get back the old value as a result.
+
+#### Hash a pointer or reference by address, not value
+
+In this release, we have introduced:
+
+- [`ptr::hash`]
+
+This function takes a raw pointer and hashes it. Using `ptr::hash`,
+you can avoid hashing the pointed-to value of a reference and instead hash the address.
+
+#### Copy the contents of an `Option<&T>`
+
+From the very beginning with Rust 1.0.0,
+the methods `Option::cloned` for `Option<&T>` and `Option<&mut T>` have allowed you to clone the contents in case of `Some(_)`.
+However, cloning can sometimes be an expensive operation and the methods `opt.cloned()` provided no hints to that effect.
+
+With this release of Rust, we introduced:
+
+- [`Option::copied`] for both `Option<&T>` and `Option<&mut T>`
+
+The functionality of `opt.copied()` is the same as for `opt.cloned()`.
+However, calling the method requires that `T: Copy`.
+Using this method, you can make sure that code stops compiling should `T` no longer implements `Copy`.
+
+### Changes in Clippy
+
+[relnotes-clippy]: 
+https://github.com/rust-lang/rust-clippy/blob/master/CHANGELOG.md#rust-135-beta
+
+[`drop_bounds`]: https://rust-lang.github.io/rust-clippy/master/index.html#drop_bounds
+
+[split_lint]: https://github.com/rust-lang/rust-clippy/pull/4101
+
+In this release of Rust,
+Clippy (a collection of lints to catch common mistakes and improve your Rust code) added a new lint [`drop_bounds`].
+This lint triggers when you add a bound `T: Drop` to a generic function. For example:
+
+```rust
+fn foo<T: Drop>(x: T) {}
+```
+
+Having a bound `T: Drop` is almost always a mistake as it excludes types,
+such as `u8`, which have trivial drop-glues.
+Moreover, `T: Drop` does not account for types like `String` not having interesting destructor behavior directly but rather as a result of embedding types,
+such as `Vec<u8>`, that do.
+
+In addition to [`drop_bounds`],
+this release of Clippy [split][split_lint] the lint`redundant_closure` into `redundant_closure` and `redundant_closure_for_method_calls`.
+
+See the [detailed release notes for Clippy][relnotes-clippy] for more details.
+
+### Changes in Cargo
```

Comment:
r? @ehuss

If you want to add any highlights please send in a patch.

Reply:
I don't think there is anything worth mentioning here for this release, so I'm r+.

Reply:
OK; cool. resolving then =)